src/route-href.js

```diff
@@ -47,6 +47,11 @@ export class RouteHref {
 
         let href = this.router.generate(this.route, this.params);
 
+        // Ensure fragments are created correctly
+        if (href.startsWith('#') && window.location) {
```

Comment:
Unfortunately, this approach will not make it in. Though in your case it does probably fix the issue, there are two key problems. First, and most importantly, we don't and can't depend on the window object as this can be run in node in a SSR scenario. Second, this code should be relegated to the `this.router.generate` function.